router/xlink_transport/config.go

```diff
@@ -55,11 +55,11 @@ func loadListenerConfig(data map[interface{}]interface{}) (*listenerConfig, erro
 		}
 	}
 
-	if value, found := data["type"]; found {
-		if typeString, ok := value.(string); ok {
-			config.linkType = typeString
+	if value, found := data["costTags"]; found {
```

Comment:
could either accept only `[]interface{}` or both `[]interface{}` and string (for single cost tag). In general we aren't that nice, so probably should do the first for consistency :)